airflow/api/common/trigger_dag.py

```diff
@@ -68,7 +68,9 @@ def _trigger_dag(
                 f"[{min_dag_start_date.isoformat()}] from DAG's default_args"
             )
 
-    run_id = run_id or DagRun.generate_run_id(DagRunType.MANUAL, execution_date)
+    run_id = run_id or dag.timetable.generate_run_id(
+        run_type=DagRunType.MANUAL, logical_date=execution_date, data_interval=None
```

Comment:
We always need a data interval anyway, so the `data_interval=None` case can be avoided by moving the data interval logic before this line.